python/hail/typ.py

```diff
@@ -545,6 +545,27 @@ def __init__(self, names, types, required = False):
 
         super(TStruct, self).__init__(jtype)
 
+    @classmethod
+    def from_fields(cls, fields, required=False):
+        """
+
+        Creates a new TStruct with input fields.
+
+        :param list of Fields fields: The TStruct fields
+        :param bool required: Whether the Struct can be missing
+        :return: TStruct wit input fields
+        :rtype: :class:`.TStruct`
+        """
+
+        struct = TStruct.__new__(cls)
+        struct.fields = fields
+        jfields = [scala_object(Env.hail().expr, 'Field').apply(
+            f.name, f.typ._jtype, i, Env.jutils().javaMapToMap(f.attributes)) for i,f in enumerate(fields)]
+        jtype = scala_object(Env.hail().expr, 'TStruct').apply(jindexed_seq(jfields), required)
+        struct._jtype = jtype
```

Comment:
I think this can be:
```python

        jfields = [scala_object(Env.hail().expr, 'Field').apply(
            f.name, f.typ._jtype, i, Env.jutils().javaMapToMap(f.attributes)) for i,f in enumerate(fields)]
        jtype = scala_object(Env.hail().expr, 'TStruct').apply(jindexed_seq(jfields), required) 
        return TStruct._from_java(jtype)
```